src/pages/Intents/intentsDatastoreQuery.js

```diff
@@ -0,0 +1,76 @@
+import { useDataMutation, useDataQuery } from '@dhis2/app-runtime'
+import isEmpty from 'lodash/isEmpty'
+import { useEffect, useState } from 'react'
+import { CUSTOM_INTENTS, NAMESPACE } from '../../constants/data-store'
+import { createKeyCustomIntent, updateSharingMutation } from '../../modules'
+
+/**
+ * update data store
+ * key: Custom intents
+ * */
+
+export const saveCustomIntentsMutation = {
+    resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}`,
+    type: 'update',
+    data: ({ settings }) => ({
+        ...settings,
+        lastUpdated: new Date().toJSON(),
+    }),
+}
+
+/**
+ * Query to get Customs Intents from Data store
+ * key: Custom Intents
+ * */
+
+export const getCustomIntentsQuery = {
+    customIntents: {
+        resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}`,
+    },
+}
+
+const getMetadataQuery = {
+    customIntents: {
+        resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}/metaData`,
+    },
+}
+
+export const useReadCustomIntentsDataStore = () => {
+    const { data, loading: loadingNamespace } = useDataQuery(
+        getCustomIntentsQuery
+    )
+    const [mutate] = useDataMutation(createKeyCustomIntent)
+    const { refetch } = useDataQuery(getMetadataQuery, { lazy: true })
+    const [mutateSharing] = useDataMutation(updateSharingMutation)
+    const [customIntents, setCustomIntents] = useState([])
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+        const fetchData = async () => {
```

Comment:
I find the name of this function a bit confusing as it's not strictly fetching data but rather setting up customIntents key if it doesn't exist?